### PR TITLE
Add cherry-picker documentation

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/README_AIRFLOW3_DEV.md [hunk 1]
```diff
@@ -122,6 +124,80 @@ Mistakes happen, and such backport PR work might fall through cracks. Therefore,
 
 This way release manager can verify (as usual) if all the "expected" PRs have been backported and cherry-pick remaining PRS.
 
+
+We are using `cherry-picker` - a [tool](https://github.com/python/cherry-picker) that has been developed by
+Python developers. It allows to easily cherry-pock PRs from one branch to another. It works both - via
```
Comment (gopidesupavan) [inline dev/README_AIRFLOW3_DEV.md]:
```suggestion
Python developers. It allows to easily cherry-pick PRs from one branch to another. It works both - via
```

File: dev/README_AIRFLOW3_DEV.md [hunk 2]
```diff
@@ -122,6 +124,80 @@ Mistakes happen, and such backport PR work might fall through cracks. Therefore,
...
+command line and via GitHub Actions interface.
+
+## How to backport PR with GitHub Actions
+
+When you want to backport commit via GitHub actions (you need to be a committer), you
+should use "Backport commit" action. You need to know the commit hash of the commit you want to backport.
+You can pin the workflow from the list of workflows for easy access to it.
+
+[!NOTE]
+It should be the commit hash of the commit in the `main` branch, not in the original PR - you can fint it
```
Comment (gopidesupavan) [inline dev/README_AIRFLOW3_DEV.md]:
```suggestion
It should be the commit hash of the commit in the `main` branch, not in the original PR - you can find it
```

**Review by jscheffl — APPROVED** [review]
Cool!
Maybe should we also just model `cherry-picker`being a dependency in breeze that this is automatically installed when setting up breeze? :-D

File: dev/README_AIRFLOW3_DEV.md [hunk 3]
```diff
@@ -122,6 +124,80 @@ Mistakes happen, and such backport PR work might fall through cracks. Therefore,
...
+
+Then, in order to backport a commit to a branch, you can use the following command:
+
+```bash
+cherry-picker COMMIT_SHA BRANCH_NAME1 [BRANCH_NAME2 ...]
+```
+
+This will create a new branch with the cherry-picked commit and open a PR against the target branch in
+your browser.
+
```
Comment (gopidesupavan) [inline dev/README_AIRFLOW3_DEV.md:178]:
```suggestion

If the GH_AUTH environment variable is set in your command line, the cherry-picker automatically creates a new pull request when there are no conflicts. To set GH_AUTH, use the token from your GitHub repository.

To set GH_AUTH run this:

```bash
export GH_AUTH={token}
```
```

Comment (gopidesupavan) [reply dev/README_AIRFLOW3_DEV.md:178]: is this fine to add? :)

**Review by gopidesupavan — APPROVED** [review]
I love the screenshots :)

**Comment by potiuk** [comment]
> Cool! Maybe should we also just model `cherry-picker`being a dependency in breeze that this is automatically installed when setting up breeze? :-D

I think that is mostly an "external" dependency that should be use outside of breeze and independently which environment you are in.  If you install it with `breeze` it will shadow the one that you have installed with `uv tool` or `pipx`. Not sure if that is what we want :)

**Pull request merged**